frontend-svelte/src/Interface.svelte

```diff
@@ -0,0 +1,113 @@
+<script>
+  import {
+    inputComponentMap,
+    outputComponentMap,
+  } from "./components/directory.js";
+
+  export let input_components;
+  export let output_components;
+  export let theme;
+  export let fn;
+
+  let input_values = input_components.map((component) =>
+    "default" in component ? component.default : null
+  );
+  let output_values = new Array(output_components.length).fill(null);
+
+  const setValues = (index, value) => {
+    input_values[index] = value;
+  };
+  const submit = () => {
+    fn("predict", { data: input_values }).then((output) => {
+      output_values = output["data"];
+    });
+  };
+  const clear = () => {};
+</script>
+
+<div class="gradio-interface" {theme}>
+  <div class="panels flex flex-wrap justify-center gap-4">
+    <div class="panel flex-1">
+      <div
+        class="component-set p-2 rounded flex flex-col flex-1 gap-2"
+        style="min-height: 36px"
+      >
+        {#each input_components as input_component, i}
+          <div class="component" key={i}>
+            <div class="panel-header mb-1.5">{input_component.label}</div>
+            <svelte:component
+              this={inputComponentMap[input_component.name]}
+              {...input_component}
+              {theme}
+              value={input_values[i]}
+              setValue={setValues.bind(this, i)}
```

Comment:
I think this would be more idiomatic as `setValue={value => setValues(i, value)}`